internal/server/auth.go

```diff
@@ -3,43 +3,10 @@ package server
 import (
 	"fmt"
 	"net/http"
-	"regexp"
-
-	"github.com/gorilla/mux"
 
 	"github.com/pomerium/datasource/internal"
 )
 
-// AuthorizationBearerMiddleware enforces Authorization: Bearer token
-func AuthorizationBearerMiddleware(token string) mux.MiddlewareFunc {
-	mw := bearerMiddleware{token}
-	return mw.middleware
-}
-
-type bearerMiddleware struct {
-	Token string
-}
-
-var (
-	reAuthorizationBearer = regexp.MustCompile(`^Bearer (\S+)$`)
-)
-
-func (mw *bearerMiddleware) middleware(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		match := reAuthorizationBearer.FindStringSubmatch(r.Header.Get("Authorization"))
-		if len(match) != 2 {
-			http.Error(w, "Authorization: Bearer token required", http.StatusUnauthorized)
-			return
-		}
-
-		if match[1] == string(mw.Token) {
```

Comment:
Sorry I missed this the first time and not relevant now but`subtle.ConstantTimeCompare` safer for side channel attacks.